testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/ClientTokenExchangeTest.java

```diff
@@ -482,9 +504,80 @@ public void testDirectImpersonation() throws Exception {
             Assert.assertTrue(response.getStatus() >= 400);
             response.close();
         }
+    }
+
+    private static void addDirectExchanger(KeycloakSession session) {
```

Comment:
Is it possible to somehow "merge" this code with the code used in the method "setupRealm" method to avoid having same/similar code on 2 places? Maybe introduce some private method (with some parameters like clientId etc if needed) and call it from both "setupRealm" and "addDirectImpersonation" ?

Reply:
I've tried to do this, but it was pretty hard. So, I'll try another way.